### PR TITLE
samples: nrf5340: multiprotocol_rpmsg: decrease BT_MAX_CONN

### PR DESCRIPTION
**Pull request by jciupis**
The multiprotocol_rpmsg is hovering around the limits of RAM available on nRF5340 network core. In order to free some of the memory and make it easier to extend the sample with additional features, this commit reduces the maximum number of concurrent BLE connections to 4. In real-life scenarios, it is unexpected for more concurrent connections to be needed*. The decrease should therefore impose no actual limitations on the users while saving a lot of RAM.

*in fact, is is uncommon for any multiprotocol application to use more concurrent connections than 2, so there's still a margin.

Signed-off-by: Jędrzej Ciupis <jedrzej.ciupis@nordicsemi.no>

### CODE REVIEW

**Review by rugeGerritsen — APPROVED** [review]

**Review by hubertmis — APPROVED** [review]
Should this be accompanied with a point in release notes?

**Comment by jciupis** [comment]
@hubertmis release notes updated.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -306,7 +306,9 @@ NFC samples
 nRF5340 samples
 ---------------
 
-|no_changes_yet_note|
+* :ref:`multiprotocol-rpmsg-sample`:
+
+  * Decreased the maximum supported number of concurrent BLE connections to 4.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:311]:
```suggestion
  * Decreased the maximum supported number of concurrent Bluetooth LE connections to 4.
```
BLE acronym is forbidden in docs.

**Review by greg-fer — CHANGES_REQUESTED** [review]
One edit.

**Review by greg-fer — APPROVED** [review]

**Pull request merged**